### ISSUE TITLE
Documentation for abstract fields?

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/sdk#43712]
It came up in an internal code review that `abstract` fields are not covered by the nnbd documentation, though they are part of the release.  Is this covered somewhere else?  If not, should we cover it as part of the null safety docs?

cc @kwalrath @munificent @eernstg

**Comment by eernstg** [comment]
So you are thinking about something more immediately accessible than https://github.com/dart-lang/language/blob/master/accepted/future-releases/abstract-external-fields/feature-specification.md?

**Comment by leafpetersen** [comment]
Yes, I pointed to that in the code review (and it's a good description!), but I think we should mention it somewhere in the language tutorials, Effective Dart, null safety documentation etc.

**Comment by munificent** [comment]
It feels weird to me to put it in the null safety docs. Maybe just in the CHANGELOG?

**Comment by leafpetersen** [comment]
> It feels weird to me to put it in the null safety docs.

It's definitely very relevant to null safety, I've already used it in one migration.  Any time you have an abstract class with a field and you intend the field to be non-nullable, the right migration (albeit a breaking one) is to use an abstract field.

**Comment by munificent** [comment]
Oh, crap, *abstract* fields. I spaced and read that as *external* fields. I forgot that we added both of them.

Yes, I think those are important enough to document in the Language Tour and other places. I like the idea of mentioning it in the null safety docs too as a recommended migration.
